### PR TITLE
ROC-5387: ReferredToMediation and  AssigningForDirections events for claimant DQ journey

### PR DESCRIPTION
**Pull request by sabahirfan**
### JIRA link (if applicable) ###
[ROC-5387](https://tools.hmcts.net/jira/browse/ROC-5387)

### Change description ###
This PR implements `ReferredToMediation` and `AssigningForDirections` events on top of DQ claimant response journey.

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [x] No

### CODE REVIEW

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/directionsquestionnaire/PilotCourt.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package uk.gov.hmcts.cmc.domain.models.directionsquestionnaire;
+
+import lombok.AllArgsConstructor;
+
+import java.util.Arrays;
+
+@AllArgsConstructor
+public enum PilotCourt {
+    EDMONTON("Birmingham"),
```
Comment (markry-kainos) [inline domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/directionsquestionnaire/PilotCourt.java]: this looks dodgy :)

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/CaseEvent.java [hunk 2]
```diff
@@ -42,6 +42,8 @@
     SUPPORT_UPDATE("SupportUpdate"),
     GENERATE_ORDER("GenerateOrder"),
     ACTION_REVIEW_COMMENTS("ActionReviewComments"),
+    ASSIGN_FOR_DIRECTIONS("AssigningForDirections"),
```
Comment (markry-kainos) [inline ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/CaseEvent.java]: minor prefer `ASSIGNING...`

File: src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/SaveClaimantResponseTest.java [hunk 3]
```diff
@@ -141,6 +150,30 @@ public void shouldSaveClaimantResponseRejection() throws Exception {
...
+        ClaimantResponse response = SampleClaimantResponse
+            .ClaimantResponseRejection.builder()
+            .buildRejectionWithDirectionsQuestionnaire();
+
+        makeRequest(claim.getExternalId(), SUBMITTER_ID, response)
+            .andExpect(status().isCreated());
+
+        Claim claimWithClaimantResponse = claimStore.getClaimByExternalId(claim.getExternalId());
+
+        assertThat(claimWithClaimantResponse.getClaimantRespondedAt().isPresent()).isTrue();
```
Comment (markry-kainos) [inline src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/SaveClaimantResponseTest.java]: minor extra line

File: src/main/java/uk/gov/hmcts/cmc/claimstore/utils/DirectionsQuestionnaireUtils.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        if (defendantResponse.getResponseType() == FULL_DEFENCE) {
+            return ((FullDefenceResponse) defendantResponse).getDirectionsQuestionnaire()
+                .orElseThrow(IllegalStateException::new)
+                .getHearingLocation().getCourtName();
+        } else if (defendantResponse.getResponseType() == PART_ADMISSION) {
+            return ((PartAdmissionResponse) defendantResponse).getDirectionsQuestionnaire()
+                .orElseThrow(IllegalStateException::new)
+                .getHearingLocation().getCourtName();
+        } else {
+            throw new IllegalStateException("No preferred court as defendant response is full admission");
```
Comment (markry-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/utils/DirectionsQuestionnaireUtils.java]: should we be throwing here? is this an illegalstate or illegalargument? this method shouldnt receive response if its a full admission

File: src/main/java/uk/gov/hmcts/cmc/claimstore/utils/DirectionsQuestionnaireUtils.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        }
+    }
+
+    private static String getClaimantHearingCourt(ClaimantResponse claimantResponse) {
+        if (claimantResponse.getType() == REJECTION) {
+            return ((ResponseRejection) claimantResponse).getDirectionsQuestionnaire()
+                .orElseThrow(IllegalStateException::new)
+                .getHearingLocation().getCourtName();
+        } else {
+            throw new IllegalStateException("No preferred court as claimant response is not rejection.");
```
Comment (markry-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/utils/DirectionsQuestionnaireUtils.java]: same as prev

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/directionsquestionnaire/PilotCourt.java (hunk 1 above)
Comment (sabahirfan) [reply domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/directionsquestionnaire/PilotCourt.java]: good catch. 👍  corrected

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/CaseEvent.java (hunk 2 above)
Comment (sabahirfan) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/CaseEvent.java]: done

**Pull request merged**